src/test/java/io/github/mapepire_ibmi/PoolTest.java

```diff
@@ -11,49 +15,273 @@
-            futuresB.add(pool.execute("values (job_name)"));
+            futures2.add(pool.execute("values (job_name)"));
         }
-        CompletableFuture<Void> allOfB = CompletableFuture.allOf(futuresB.toArray(new CompletableFuture[0]));
-        allOfB.join();
+        CompletableFuture<Void> allFutures2 = CompletableFuture.allOf(futures2.toArray(new CompletableFuture[0]));
+        allFutures2.join();
 
-        List<String> jobNamesB = new ArrayList<>();
-        for (CompletableFuture<QueryResult<Object>> future : futuresB) {
-            jobNamesB.add(((HashMap<String, String>) future.get().getData().get(0)).get("00001"));
+        List<String> jobNames2 = new ArrayList<>();
+        for (CompletableFuture<QueryResult<Object>> future : futures2) {
+            jobNames2.add(((HashMap<String, String>) future.get().getData().get(0)).get("00001"));
         }
 
-        assertEquals(15, jobNamesB.size());
+        assertEquals(15, jobNames2.size());
         assertTrue(pool.getActiveJobCount() >= 3);
         assertTrue(pool.getActiveJobCount() <= 5);
 
         pool.end();
     }
+
+    @Test
+    void invalidSizes() throws Exception {
+        ClientException e1 = assertThrowsExactly(ClientException.class, () -> {
+            PoolOptions options = new PoolOptions(MapepireTest.getCreds(), 0, 10);
+            Pool pool = new Pool(options);
+
+            try {
+                pool.init().get();
+            } catch (Exception ex) {
+                pool.end();
+                throw ex;
+            }
+        });
+
+        ClientException e2 = assertThrowsExactly(ClientException.class, () -> {
+            PoolOptions options = new PoolOptions(MapepireTest.getCreds(), 10, 0);
+            Pool pool = new Pool(options);
+
+            try {
+                pool.init().get();
+            } catch (Exception ex) {
+                pool.end();
+                throw ex;
+            }
+        });
+
+        ClientException e3 = assertThrowsExactly(ClientException.class, () -> {
+            PoolOptions options = new PoolOptions(MapepireTest.getCreds(), 1, 10);
+            Pool pool = new Pool(options);
+
+            try {
+                pool.init().get();
+            } catch (Exception ex) {
+                pool.end();
+                throw ex;
+            }
+        });
+
+        assertEquals("Max size must be greater than 0", e1.getMessage());
+        assertEquals("Max size must be greater than 0", e2.getMessage());
+        assertEquals("Max size must be greater than or equal to starting size", e3.getMessage());
+    }
+
+    @Test
+    void performance() throws Exception {
+        long startPool1 = System.currentTimeMillis();
+        PoolOptions options = new PoolOptions(MapepireTest.getCreds(), 5, 3);
+        Pool pool = new Pool(options);
+        pool.init().get();
+
+        List<CompletableFuture<QueryResult<Object>>> futures1 = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < 20; i++) {
```
We need to test the same number of statements for each pool

Reply:
Thanks for catching that. I believe this test will fail because of https://github.com/Mapepire-IBMi/mapepire-java/issues/43. It will be fixed in separate PR